### ISSUE TITLE
Add way to set ID for a localization string?

### ISSUE THREAD

**Issue by jruales** [microsoft/vscode#193738]
Currently, each string is assigned an ID based on a hash of the English string and the comment. The problem with this is that any small change to a string or its comment will change the ID, requiring a new translation from scratch. It might be good to allow the user to specify an ID to use for a given English string. That way, if the user makes a small change to the English string, the ID will remain the same, and the translator will be able to use the old translation as a basis for the new translation, only changing the small part that was modified in the English string.

**Comment by ehbales** [comment]
I agree with the points made by @rudyflores. Another factor to consider is that duplications are sometimes required. For example, I checked with one of our accessibility testers and she confirmed that aria labels often duplicate text that displays in the UI. 
Thanks for considering support for key IDs!

**Comment by RichardLuo0** [comment]
Any updates on this? I can't believe this can take that long. I can't use l10n because of this.
